### ISSUE TITLE
Counters with spaces in name are not enabled.

### ISSUE THREAD

**Issue by philkild** [open-telemetry/opentelemetry-dotnet#4117]
List of all OpenTelemetry NuGet
packages and version that you are
using (e.g. OpenTelemetry 1.0.2):

OpenTelemetry 1.4.0-rc2
OpenTelemetry.API 1.4.0-rc2
OpenTelemetry.Exporter.Console 1.4.0-rc2

Runtime version (e.g. net462, net48, netcoreapp3.1, net6.0 etc. You can
find this information from the *.csproj file):

net7.0

---

## Issue

Creating a counter that has a space in the name results in the counter not being enabled with no indication as to why...  Please can some sort of validation be added to make this more obvious if no spaces is a requirement.

![image](https://user-images.githubusercontent.com/20355325/215334313-079a7ae8-5dd4-4863-bc10-f5c1605e320a.png)

**Comment by cijothomas** [comment]
https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/api.md#instrument-naming-rule
Metric names must follow the rules from above, else they will not be enabled.

There is internal logging when any instrument is disabled, so you can use that to troubleshoot: https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/src/OpenTelemetry/README.md#troubleshooting

It is not possible to throw exceptions from the APIs, as the general guidelines for telemetry APIs is to *not* throw errors at runtime.

**Comment by philkild** [comment]
I get that it shouldn't throw but it feels like there should be a property on the counter for state or reason etc. to help developers. The first thing I did when I realised it wasn't working was to check all the properties to see what was wrong (given no exception) so it would have saved me a load of time and I didn't know about the logging. 

At the very least, I hope people that have the same issue as me find this post and save themselves some time...
